synapse/handlers/oidc_handler.py

```diff
@@ -172,6 +250,31 @@ async def load_jwks(self, force=False) -> Dict[str, List[Dict[str, str]]]:
         return jwk_set
 
     async def _exchange_code(self, code: str) -> Dict[str, str]:
+        """Exchange an authorization code for a token.
+
+        This calls the ``token_endpoint`` with the authorization code we
+        received in the callback to exchange it for a token. The call uses the
+        ``ClientAuth`` to authenticate with the client with its ID and secret.
+
+        Args:
+            code (str): The autorization code we got from the callback.
+
+        Returns:
+            dict: contains various tokens.
+
+            May look like this::
+
+                {
+                    'token_type': 'bearer',
+                    'access_token': 'abcdef',
+                    'expires_in': 3599,
```

Comment:
Doesn't this mean the return type is _not_ `Dict[str, str]`?

Reply:
I think the return type should likely be `JsonDict` (found in `synapse.types`). It seems like there's a handful of other spots here that should also be using this instead of `Dict[str, str]`.

Reply:
Hm. I would've imagined using a `typing_extensions.TypedDict` would work here, but I guess there's no need to if it's only being used as part of a response.

Reply:
Did not know about typing_extensions ; I wanted to use a TypedDict in the beginning, but that would not have worked with py3.5. That's done.